java/source/data.rst

```diff
@@ -0,0 +1,360 @@
+    vec.set(0, "ba".getBytes());
+    vec.set(1, "abc".getBytes());
+    vec.set(2, "aa".getBytes());
+
+    VectorValueComparator<VarCharVector> comparatorValues = new TestVectorValueComparator();
+    VectorValueComparator<VarCharVector> stableComparator = new StableVectorComparator<>(comparatorValues);
+    stableComparator.attachVector(vec);
+
+    System.out.println(stableComparator.compare(0, 1) > 0);
+    System.out.println(stableComparator.compare(1, 2) < 0);
+
+.. testoutput::
+
+    true
+    true
+
+Search Values on the Array
+==========================
+
+Linear Search - O(n)
+********************
+
+Algorithm: org.apache.arrow.algorithm.search.VectorSearcher#linearSearch - O(n)
+
+.. testcode::
+
+    import org.apache.arrow.algorithm.search.VectorSearcher;
+    import org.apache.arrow.algorithm.sort.DefaultVectorComparators;
+    import org.apache.arrow.algorithm.sort.VectorValueComparator;
+    import org.apache.arrow.vector.IntVector;
+    import org.apache.arrow.memory.RootAllocator;
+
+    RootAllocator rootAllocator = new RootAllocator(Long.MAX_VALUE);
+    IntVector linearSearchVector = new IntVector("linearSearchVector", rootAllocator);
+    linearSearchVector.allocateNew(10);
+    linearSearchVector.setValueCount(10);
+    for (int i = 0; i < 10; i++) {
+        linearSearchVector.set(i, i);
+    }
+    VectorValueComparator<IntVector> comparatorInt = DefaultVectorComparators.createDefaultComparator(linearSearchVector);
+    List<Integer> listResultLinearSearch = new ArrayList<Integer>();
+    for (int i = 0; i < 10; i++) {
+       int result = VectorSearcher.linearSearch(linearSearchVector, comparatorInt, linearSearchVector, i);
+       listResultLinearSearch.add(result);
+    }
+
+    System.out.println(listResultLinearSearch);
+
+.. testoutput::
+
+    [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+Binary Search - O(log(n))
+*************************
+
+Algorithm: org.apache.arrow.algorithm.search.VectorSearcher#binarySearch - O(log(n))
+
+.. testcode::
+
+    import org.apache.arrow.algorithm.search.VectorSearcher;
+    import org.apache.arrow.algorithm.sort.DefaultVectorComparators;
+    import org.apache.arrow.algorithm.sort.VectorValueComparator;
+    import org.apache.arrow.vector.IntVector;
+    import org.apache.arrow.memory.RootAllocator;
+
+    RootAllocator rootAllocator = new RootAllocator(Long.MAX_VALUE);
+    IntVector binarySearchVector = new IntVector("", rootAllocator);
+    binarySearchVector.allocateNew(10);
+    binarySearchVector.setValueCount(10);
+    for (int i = 0; i < 10; i++) {
+        binarySearchVector.set(i, i);
+    }
+    VectorValueComparator<IntVector> comparatorInt = DefaultVectorComparators.createDefaultComparator(binarySearchVector);
+    List<Integer> listResultBinarySearch = new ArrayList<Integer>();
+    for (int i = 0; i < 10; i++) {
+       int result = VectorSearcher.binarySearch(binarySearchVector, comparatorInt, binarySearchVector, i);
+       listResultBinarySearch.add(result);
+    }
```

Comment:
I think that the for loop to search for all values makes things more complicate to understand. Let's stick to how you search 1 value, it will make the recipe easier to understand.

Reply:
Changed